### PR TITLE
ENG-7700: Improve storage documentation when working with data science pipelines 

### PR DESCRIPTION
**Pull request by chtyler**
## Description
Module written on how pipeline data is stored when running a pipeline in RHOAI. The location of the storage depends on where you ran the pipeline from. For example, running a pipeline from the dashboard results in a different directory structure than running a pipeline from Elyra. 

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: modules/storing-data-with-data-science-pipelines.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+:_module-type: REFERENCE
+
+[id="storing-data-with-data-science-pipelines_{context}"]
+= Storing data with data science pipelines
+
+[role="_abstract"]
+When you run a data science pipeline, {productname-short} stores the pipeline YAML configuration file and resulting pipeline run artifacts in your storage bucket `root` directory. The directories that contain pipeline run artifacts can differ depending on where you executed the pipeline run from. See the following table for further information:
```
Comment (grainnejenningsRH) [inline modules/storing-data-with-data-science-pipelines.adoc:7]: Suggest updating to 'in the `root` directory of your storage bucket.'

File: modules/storing-data-with-data-science-pipelines.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+.Pipeline configuration file and artifacts storage locations
+[cols="3]
+|===
+| Pipeline run source | Pipeline storage directory | Run artifacts storage directory
+
+| Ran from the {productname-short} dashboard
```
Comment (grainnejenningsRH) [inline modules/storing-data-with-data-science-pipelines.adoc:14]: Could probably remove 'Ran from the' from this line and line 22 without sacrificing any clarity.

File: modules/storing-data-with-data-science-pipelines.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+Example: `iris-training-pipeline/2g48k8pw-a8ib-4884-9145-h41j7599h3ds`
+
+| Ran from the JupyterLab Elyra extension
+| `/pipelines/<pipeline_version_id>`
+| `/<pipeline_name_timestamp>` 
+
+Example: `/hello-generic-world-0523161704`
+
+In the JupyterLab Elyra extension, you can also set an link:https://elyra.readthedocs.io/en/latest/user_guide/pipelines.html#generic-node-properties[object storage path prefix].
```
Comment (grainnejenningsRH) [inline modules/storing-data-with-data-science-pipelines.adoc:28]: Are users working within an extension here, or would it be more accurate to say 'With the JupyterLab Elyra extension'?

File: modules/storing-data-with-data-science-pipelines.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+|===
+| Pipeline run source | Pipeline storage directory | Run artifacts storage directory
+
+| Ran from the {productname-short} dashboard
+| `/pipelines/<pipeline_version_id>` 
+
+Example: `/pipelines/1d01c4eb-d2ab-4916-9935-a73a5580f1fb`
+| `/<pipeline_name>/<pipeline run_id>` 
+
+Example: `iris-training-pipeline/2g48k8pw-a8ib-4884-9145-h41j7599h3ds`
```
Comment (grainnejenningsRH) [inline modules/storing-data-with-data-science-pipelines.adoc:20]: Are the strings in the storage directory paths just random unique strings per pipeline?

Comment (chtyler) [reply modules/storing-data-with-data-science-pipelines.adoc:20]: Yes they are random unique strings per pipeline and per pipeline run

File: modules/storing-data-with-data-science-pipelines.adoc (hunk 3 above)
Comment (chtyler) [reply modules/storing-data-with-data-science-pipelines.adoc:28]: They are effectively in JupyerLab working with the extension. I take your point though. I can change it,.

File: modules/storing-data-with-data-science-pipelines.adoc (hunk 2 above)
Comment (chtyler) [reply modules/storing-data-with-data-science-pipelines.adoc:14]: agreed

**Review by grainnejenningsRH — APPROVED** [review]

**Comment by jgarciao** [comment]
lgtm

**Pull request merged**